From 6f2846a1e927710a2a7aa72cce47ea5c841e4417 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Mon, 16 Apr 2018 11:33:11 +0300
Subject: [PATCH 546/706] bus: fsl-mc: define macros for iommu enablement for
 fsl-mc bus

Macros to get coherency and the container device of the devices on
fsl-mc bus are required to suport SMMU for this bus. This patch
defines the same.

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
(cherry picked from commit fb71cd6c0635be04dabe6a84cbd103cb1d6a73bb)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 include/linux/fsl/mc.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/linux/fsl/mc.h b/include/linux/fsl/mc.h
index 29398cf2fcf3..179af7d21431 100644
--- a/include/linux/fsl/mc.h
+++ b/include/linux/fsl/mc.h
@@ -358,6 +358,18 @@ int mc_send_command(struct fsl_mc_io *mc_io, struct fsl_mc_command *cmd);
 #define dev_is_fsl_mc(_dev) (0)
 #endif
 
+/* Macro to check if a device is a container device */
+#define fsl_mc_is_cont_dev(_dev) (to_fsl_mc_device(_dev)->flags & \
+	FSL_MC_IS_DPRC)
+
+/* Macro to get the container device of a MC device */
+#define fsl_mc_cont_dev(_dev) (fsl_mc_is_cont_dev(_dev) ? \
+	(_dev) : (_dev)->parent)
+
+#define fsl_mc_is_dev_coherent(_dev) \
+	(!((to_fsl_mc_device(_dev))->obj_desc.flags & \
+	FSL_MC_OBJ_FLAG_NO_MEM_SHAREABILITY))
+
 /*
  * module_fsl_mc_driver() - Helper macro for drivers that don't do
  * anything special in module init/exit.  This eliminates a lot of
-- 
2.17.1

